utils/validation.js

```diff
@@ -181,6 +181,42 @@ const getToken = async (req, res) => {
     }
 }
 
+const processMouthwashEligibility = (data) => {
+    // Conditions for initialized: baselineMouthwashSample is no, bloodOrUrineCollected is yes, 
+    // kitStatus does not yet have a value, processParticipantHomeMouthwashKitData passes
+    const updates = {};
+    const {processParticipantHomeMouthwashKitData} = require('./firestore');
+    if(
+        data[conceptIds.withdrawConsent] == conceptIds.no &&
+        data[conceptIds.participantDeceasedNORC] == conceptIds.no &&
+        data[conceptIds.activityParticipantRefusal] && data[conceptIds.activityParticipantRefusal][conceptIds.baselineMouthwashSample] == conceptIds.no &&
+        data[conceptIds.collectionDetails] && data[conceptIds.collectionDetails][conceptIds.baseline] &&
+        data[conceptIds.collectionDetails][conceptIds.baseline][conceptIds.bloodOrUrineCollected] == conceptIds.yes &&
+        data[conceptIds.collectionDetails][conceptIds.baseline][conceptIds.bloodOrUrineCollectedTimestamp] >= '2024-04-01T00:00:00.000Z' &&
+        (
+            !data[conceptIds.collectionDetails][conceptIds.baseline][conceptIds.bioKitMouthwash] ||
+            data[conceptIds.collectionDetails][conceptIds.baseline][conceptIds.bioKitMouthwash][conceptIds.kitStatus]
+        )
+    ) {
+        const isEligible = !!processParticipantHomeMouthwashKitData(data, true);
+        if(isEligible) {
+            updates[`${conceptIds.collectionDetails}.${conceptIds.baseline}.${conceptIds.bioKitMouthwash}.${conceptIds.kitStatus}`] = conceptIds.initialized;
+        }
+    } else if(
```

Comment:
Would converting this to an `if` block (instead of `else if`)and using the updates object:
`updates[conceptIds.collectionDetails][conceptIds.baseline][conceptIds.bioKitMouthwash]`
eliminate the need for the second `processMouthwashEligibility` call (outside the transaction) in `submitSpecimen`?

I'm just curious when this `else if` case is applicable. When is kitStatus is set to initialized and it needs to be reverted?

Reply:
The else if case is in case the participant was previously eligible but updated to be ineligible due to changing their address to a PO box or something. Somewhat of an edge case but I figured it was best to include it.